appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/visitor/OpenApiWalker.java

```diff
@@ -104,211 +95,161 @@
-
             for (org.glassfish.hk2.classmodel.reflect.Parameter parameter : method.getParameters()) {
-                processAnnotation(parameter, annotationClass, annotationFunction, annotations, context, alternatives);
+                processAnnotation((E) parameter, annotations, visitor, new OpenApiContext(context, method));
             }
         }
     }
 
     @SuppressWarnings("unchecked")
-    @SafeVarargs
-    private static <A extends Annotation, E extends AnnotatedElement> void processAnnotation(
-            AnnotatedElement element,
-            Class<A> annotationClass,
-            VisitorFunction<AnnotationModel, E> annotationFunction,
-            AnnotationInfo annotations,
-            ApiContext context,
-            Class<? extends Annotation>... alternatives
-    ) {
-        // If it's just the one annotation class
-        // Check the element
-        if (annotations.isAnnotationPresent(annotationClass, element)) {
-            annotationFunction.apply(annotations.getAnnotation(annotationClass, element), (E) element, context);
-        } else if (element instanceof MethodModel && annotations.isAnnotationPresent(annotationClass)
-                && !annotations.isAnyAnnotationPresent(element, alternatives)) {
-            // If the method isn't annotated, inherit the class annotation
-            if (context.getPath() != null) {
-                annotationFunction.apply(annotations.getAnnotation(annotationClass), (E) element, context);
-            }
-        }
-    }
-
-    /**
-     * Generates a map listing the location each resource class is mapped to.
-     */
-    private Map<String, Set<Type>> generateResourceMapping() {
-        Set<Type> classList = new HashSet<>();
-        Map<String, Set<Type>> mapping = new HashMap<>();
-        for (Type type : allowedTypes) {
-            if(type instanceof ClassModel) {
-                ClassModel classModel = (ClassModel) type;
-                if(classModel.getAnnotation(ApplicationPath.class.getName()) != null) {
-                    // Produce the mapping
-                    AnnotationModel annotation = classModel.getAnnotation(ApplicationPath.class.getName());
-                    String key = annotation.getValue("value", String.class);
-                    Set<Type> resourceClasses = new HashSet<>();
-                    mapping.put(key, resourceClasses);
-                    try {
-                        Class<?> clazz = appClassLoader.loadClass(classModel.getName());
-                        Application app = (Application) clazz.newInstance();
-                        // Add all classes contained in the application
-                        resourceClasses.addAll(app.getClasses()
-                                .stream()
-                                .map(Class::getName)
-                                .filter(name -> !name.startsWith("org.glassfish.jersey")) // Remove all Jersey providers
```

Comment:
I assume filtering is no longer needed because of the new way we extract class information or it is done elsewhere now?

Reply:
Added here:
https://github.com/payara/Payara/blob/8d65b9ebcbfd989d06862e1661e10d5385695482/appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/visitor/OpenApiContext.java#L166